test/tests/amqp/test_amqp_node_rediscover.py

```diff
@@ -0,0 +1,537 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+Author(s):
+Norton Luo
+
+'''
+
+from time import sleep
+import Queue
+import random
+import socket
+import flogging
+import logging
+import pika
+import unittest
+import json
+import threading
+from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
+import fit_common
+import test_api_utils
+from nose.plugins.attrib import attr
+logs = flogging.get_loggers()
+amqp_message_received = False
+amqp_queue = Queue.Queue(maxsize=0)
+node_uuid = ""
+nodefound_id = ""
+ebhook_received = False
+webhook_body = ""
+
+
+class AmqpWorker(threading.Thread):
+    '''
+    This AMQP worker Class will creat another thread when initialized and runs asynchronously.

```

Comment:
I know that you've synced with other guys that it will have refactoring works in this parts in future, please also have a note 'TODO' in annotation for the similar parts so that the developers won't be confused about the duplicated codes.